service/datasync/exec_data_sync.go

```diff
@@ -0,0 +1,194 @@
+package datasync
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/onflow/flow-go/consensus/hotstuff/model"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow/protobuf/go/flow/entities"
+	execData "github.com/onflow/flow/protobuf/go/flow/executiondata"
+	"github.com/optakt/flow-dps/models/dps"
+	"github.com/rs/zerolog"
+	"google.golang.org/grpc"
+	"math"
+	"sync/atomic"
+
+	"cloud.google.com/go/storage"
+	"github.com/fxamacker/cbor/v2"
+
+	"github.com/onflow/flow-archive/models/archive"
+)
+
+// ExecDataSync is a component to retrieve execution data from a trusted access node API instead of the GCPStreamer
+type ExecDataSync struct {
+	log         zerolog.Logger
+	decoder     cbor.DecMode
+	execDataApi execData.ExecutionDataAPIClient
+	queue       *archive.SafeDeque // queue of block identifiers for next downloads
+	buffer      *archive.SafeDeque // queue of downloaded execution data records
```

Comment:
these names are a little confusing since they don't reflect what they hold. maybe something like this?
```suggestion
	blocks       *archive.SafeDeque // queue of block identifiers for next downloads
	records     *archive.SafeDeque // queue of downloaded execution data records
```